src/wp-admin/includes/menu.php

```diff
@@ -368,7 +368,17 @@ function sort_menu( $a, $b ) {
 }
 unset( $last_menu_key );
 
-if ( ! user_can_access_admin_page() ) {
+if ( ! admin_page_exists() ) {
+
+	/**
+	 * Fires when access to an admin page which does not exist.
+	 *
+	 * @since 4.9.0
```

Comment:
Suggest changing to 6.9.0 to match the milestone set on the Trac ticket.

```suggestion
	 * @since 6.9.0
```